### ISSUE TITLE
[Performance] PersonaPlex performs 17 synchronized host round trips per generated frame

### ISSUE THREAD

**Issue by chaofengw-nv** [NVIDIA/TensorRT-Model-Connect#504]
## Summary

PersonaPlex warm speech-to-speech inference performs one temporal engine call plus 16 depth engine calls per generated frame, and every call synchronizes and round-trips logits through host memory.

On an NVIDIA GB300 with TensorRT 11.2 and CUDA 13.0, using TRTMC commit `935243bba029d23463910817d14edd1c1bd1d697` and `nvidia/personaplex-7b-v1` revision `fdaf4090a61cb315c138a1faee287ffd6c716309`:

- TRTMC complete-task p50: **3,580.18 ms**
- Official HF eager complete-task p50: **3,160.70 ms**
- TensorRT engine execution: **692.40 ms/request**
- Non-engine time: **2,887.78 ms/request**
- GPU utilization: **18% p50**
- Quality: model-owned task evaluation passes

The timing boundary is aligned: decoded PCM through Mimi encode, temporal/depth generation, Mimi decode, host PCM materialization, and final synchronization. Loading and file I/O are excluded.

## Diagnosis

The 66-frame workload performs approximately 1,139 temporal/depth engine launches per request. The engines account for only about 19% of complete-task latency.

The current implementation:

- synchronizes after each temporal and depth engine call;
- downloads every non-external output after each call;
- copies temporal/depth logits into CPU vectors;
- performs greedy/top-k token selection on CPU before the next dependent depth call.

A frame-count sweep confirms the cost scales with the host round trips. Increasing generated output from 54 to 66 frames adds 204 temporal/depth calls and about 634 ms of complete-task latency; only about 116 ms is additional engine execution. The remaining roughly 518 ms is about 2.5 ms per added synchronized call.

The 16 depth steps are autoregressively dependent, but that dependency does not require a host synchronization and full-logit D2H after every engine.

## Reproduction

1. Build `nvidia/personaplex-7b-v1` at the revision above in FP16 with the model-owned FP32 selector.
2. Run the `personaplex-7b-l0` speech-to-speech workload with the model-owned decoded PCM input, seed 42, and greedy token selection.
3. Use 5 complete-task warmups and 50 timed requests. Measure decoded PCM to host-visible output PCM and synchronize after every task.
4. Enable per-engine timing and record temporal/depth launch counts and complete-task wall latency.
5. Repeat with output-tail budgets that produce 54 and 66 frames while holding the input fixed.

The model-owned quality command is:

```bash
python3 tools/task_eval.py eval --model personaplex-7b-l0 --force-build --force-hf --single-device-only
```

## Expected behavior

- Keep temporal/depth cache, hidden state, and selected tokens device-resident across the generation loop.
- Use device-side argmax/top-k sampling and asynchronous same-stream chaining between dependent depth calls.
- Avoid downloading full logits and synchronizing after every temporal/depth engine.
- Copy only final user-visible outputs to host where possible.
- Preserve the official token snapshot, output PCM contract, deterministic seed behavior, and task-eval quality.
- Report engine execution, aligned complete-task p50, and GPU utilization after the change.
